### PR TITLE
feat: Implement `resetLimit`

### LINKED ISSUE #2868: [HBAR Rate Limit Redesign] Implement resetLimit
Create the **_resetLimit_** method (as per the diagram in the design doc) in the HbarLimit class, implementing the logic to reset the `spentToday` value for an `HbarSpendingPlan`

### PR DESCRIPTION
**Pull request by victor-yanev**
**Description**:

Changes:
- `HbarSpendingPlanRepository`
  - Added `resetAllSpentTodayEntries` method.
  - Updated relevant unit tests to cover the new functionality.
  - Enhanced logging for better traceability.
- `HbarLimitService`
  - Implemented `resetLimiter` method
    - it resets the `spentToday` entry for all spending plans
    - resets the `remainingBudget`
    - resets the `remainingBudgetGauge` metrics
    - resets the `dailyUniqueSpendingPlansCounter` and `averageDailySpendingPlanUsagesGauge` metrics
    - sets the new reset date to the next day at midnight
 
**Related issue(s)**:

Fixes #2868 

**Notes for reviewer**:

- Ensure that the used cache operations are efficient and do not introduce performance bottlenecks.
- Verify that the logging provides sufficient information for debugging purposes.

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/src/lib/services/hbarLimitService/index.ts [hunk 1]
```diff
@@ -140,11 +140,24 @@ export class HbarLimitService implements IHbarLimitService {
...
+    for (const subscriptionType of Object.values(SubscriptionType)) {
+      this.dailyUniqueSpendingPlansCounter[subscriptionType].reset();
+      this.averageDailySpendingPlanUsagesGauge[subscriptionType].reset();
+    }
+    const tomorrow = new Date(Date.now() + HbarLimitService.ONE_DAY_IN_MILLIS);
+    this.reset = new Date(tomorrow.setHours(0, 0, 0, 0));
+    this.logger.trace(
+      `${requestIdPrefix} HBAR Rate Limit reset: remainingBudget=${this.remainingBudget}, newResetTimestamp=${this.reset}`,
+    );
   }
```
Comment (ebadiere) [inline packages/relay/src/lib/services/hbarLimitService/index.ts:161]: Nit.  Maybe for readability:
```
  /**
   * Resets the {@link HbarSpendingPlan#spentToday} field for all existing plans.
   * @param {string} [requestId] - An optional unique request ID for tracking the request.
   * @returns {Promise<void>} - A promise that resolves when the operation is complete.
   */
  async resetLimiter(requestId?: string): Promise<void> {
    const requestIdPrefix = this.formatRequestIdMessage(requestId);
    this.logger.trace(`${requestIdPrefix} Resetting HBAR rate limiter...`);

    await this.resetRepository();
    this.resetBudget();
    this.resetMetrics();
    this.updateResetTimestamp();

    this.logger.trace(
      `${requestIdPrefix} HBAR Rate Limit reset: remainingBudget=${this.remainingBudget}, newResetTimestamp=${this.reset}`,
    );
  }

  private async resetRepository(): Promise<void> {
    await this.hbarSpendingPlanRepository.resetAllSpentTodayEntries();
  }

  private resetBudget(): void {
    this.remainingBudget = this.totalBudget;
    this.hbarLimitRemainingGauge.set(this.remainingBudget);
  }

  private resetMetrics(): void {
    for (const subscriptionType of Object.values(SubscriptionType)) {
      this.dailyUniqueSpendingPlansCounter[subscriptionType].reset();
      this.averageDailySpendingPlanUsagesGauge[subscriptionType].reset();
    }
  }

  private updateResetTimestamp(): void {
    const tomorrow = new Date(Date.now() + HbarLimitService.ONE_DAY_IN_MILLIS);
    this.reset = new Date(tomorrow.setHours(0, 0, 0, 0));
  }

  private formatRequestIdMessage(requestId?: string): string {
    return requestId ? `[RequestId: ${requestId}]` : '';
  }
  ```

**Review by ebadiere — APPROVED** [review]
Good work.

File: packages/relay/src/lib/services/hbarLimitService/index.ts (hunk 1 above)
Comment (victor-yanev) [reply packages/relay/src/lib/services/hbarLimitService/index.ts:161]: Nice one, I like the suggestion.

File: packages/relay/src/lib/services/hbarLimitService/index.ts [hunk 2]
```diff
@@ -133,18 +133,24 @@ export class HbarLimitService implements IHbarLimitService {
...
 
   /**
    * Resets the {@link HbarSpendingPlan#spentToday} field for all existing plans.
+   * @param {string} [requestId] - An optional unique request ID for tracking the request.
    * @returns {Promise<void>} - A promise that resolves when the operation is complete.
    */
-  async resetLimiter(): Promise<void> {
-    // TODO: Implement this with https://github.com/hashgraph/hedera-json-rpc-relay/issues/2868
-    throw new Error('Not implemented');
+  async resetLimiter(requestId?: string): Promise<void> {
```
Comment (konstantinabl) [inline packages/relay/src/lib/services/hbarLimitService/index.ts:144]: In the context of recent changes, shouldnt requestId be required or should I change it in my PR?

Comment (victor-yanev) [reply packages/relay/src/lib/services/hbarLimitService/index.ts:144]: Let's do this as part of https://github.com/hashgraph/hedera-json-rpc-relay/pull/2939, which would have to change this method to use a `RequestDetails` object anyway.

**Review by ebadiere — DISMISSED** [review]
LG

**Review by konstantinabl — DISMISSED** [review]

**Review by konstantinabl — APPROVED** [review]

**Pull request merged**